source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -419,14 +424,50 @@ def findExtraOverlayClasses(obj, clsList):
 		clsList.append(consoleUIAWindow)
 
 
-class WinTerminalUIA(EnhancedTermTypedCharSupport):
+class DiffBasedWinTerminalUIA(EnhancedTermTypedCharSupport):
```

Comment:
should these classes be marked as internal?

Reply:
No, for symmetry with the rest of the module.

Reply:
What do you mean by symmetry?

Reply:
The other `Win*UIA` classes in this module don't have underscores.

Reply:
I don't think that it is too late to fix the pattern, if these should be marked as internal, we should mark them.

Reply:
This would be a significant divergence from other similar classes in `NVDAObjects.UIA`. A few examples:

* `NVDAObjects.UIA.VisualStudio.IntelliSenseList`
* `NVDAObjects.UIA.wordDocument.UIACustomAttributeID`
* `NVDAObjects.UIA.wordDocument.ElementsListDialog`

Reply:
I think this is the correct divergence to make. 
What do you think @feerrenrut ? Note earlier conversation from https://github.com/nvaccess/nvda/pull/14047#issuecomment-1288244324

Reply:
I can acknowledge this is inconsistent with the existing code, however the inconsistency is due to a change process around add-on APIs. We need to be able to delineate where we commit to a stable add-on API and where we don't. If there is no expectation of add-ons needing the new code, it is safest to mark it internal only. This allows it to change more easily without going through a deprecation process. If add-on authors identify the code as useful for their add-on, and want it exposed, then we can have a discussion around that and consider if it should be exposed directly or via some level of abstraction.